src/server/frontend_wayland/data_device.cpp

```diff
@@ -320,17 +307,12 @@ void DataDevice::focus_on(wl_client* focus)
 
     if (has_focus && current_source && !current_offer)
     {
-        wl_resource* new_resource = wl_resource_create(
-            client,
-            &mw::wl_data_offer_interface_data,
-            wl_resource_get_version(resource),
-            0);
-        current_offer = new DataOffer{new_resource, current_source, this};
+        current_offer = new DataOffer{client, resource, current_source, this};
     }
 }
 
-DataOffer::DataOffer(wl_resource* new_resource, DataSource* source, DataDevice* device) :
-    mw::DataOffer(new_resource, Version<3>()),
+DataOffer::DataOffer(wl_client* client, wl_resource* parent, DataSource* source, DataDevice* device) :
+    mw::DataOffer(client, parent, Version<3>()),
```

Comment:
Why pass in the parent resource? Why not just use the one on `device`?

Reply:
OK, I _could_ use `device->client` & `device->resource`